### PR TITLE
Photo history, and significantly less repeat images

### PR DESCRIPTION
**Pull request by robacarp**

### CODE REVIEW

File: web_extension/content/scripts/new_tab.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  const feed_url = options.feed.url
-  const response = await fetch(feed_url)
-  const feed = await response.json()
-  const number = parseInt(Math.random() * feed.items.length)
-  const item = feed.items[number]
-
-  set_image(item.url)
-  show_info(item)
+async function chooseImage() {
+  let item = await PhotoChooser.pick()
```
Comment (JoshAshby) [inline web_extension/content/scripts/new_tab.js]: CONST

File: web_extension/content/scripts/photo_chooser.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    // Randomly choose one of the images which has been seen the least.
+    const number = parseInt(Math.random() * seen_the_least.length)
+    return seen_the_least[number]
+  }
+
+  static async pick () {
+    let options = new Options()
+    await options.read()
+
+    let feed = await fetch(options.feed.url)
```
Comment (JoshAshby) [inline web_extension/content/scripts/photo_chooser.js:40]: mixing `async/await` with `.then`ing :|

File: web_extension/content/scripts/new_tab.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+  if (item) {
```
Comment (JoshAshby) [inline web_extension/content/scripts/new_tab.js]:
```suggestion
  if (!item) return
```

**Review by JoshAshby — APPROVED** [review]

File: web_extension/content/scripts/photo_chooser.js (hunk 2 above)
Comment (robacarp) [reply web_extension/content/scripts/photo_chooser.js:40]: It was this or make 3 different consts. I blame JavaScript for my crimes.

File: web_extension/content/scripts/new_tab.js (hunk 3 above)
Comment (robacarp) [reply web_extension/content/scripts/new_tab.js]: Seems legit but if I take your suggestion the code won’t work anymore...the diff doesn’t produce valid codes

Comment (robacarp) [reply web_extension/content/scripts/new_tab.js]: also I miss ruby

**Pull request merged**